src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -438,7 +479,9 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		this._onCommandExecuted.fire();
 	}
 
-	private _handleCommandExecutedWindows(): void {
+	private async _handleCommandExecutedWindows(): Promise<void> {
+		await this._commandStartedWindowsBarrier?.wait();
+		this._commandExecutedWindowsBarrier = new Barrier();
```

Comment:
I don't think this one does anything as everything between it and the `open` is synchronous.